spdx-utils/src/main/kotlin/model/SpdxFile.kt

```diff
@@ -0,0 +1,193 @@
+     * and/or authors who may not be copyright holders, yet contributed to the file content.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val fileContributors: List<String>,
+
+    /**
+     * This field is deprecated since SPDX 2.0 in favor of [SpdxDocument.relationships].
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val fileDependencies: List<String> = emptyList(),
+
+    /**
+     * The name of the file.
+     */
+    @JsonProperty("fileName")
+    val filename: String,
+
+    /**
+     * The types of the file.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val fileTypes: List<Type> = emptyList(),
+
+    /**
+     * Any relevant background references or analysis that went in to arriving at the Concluded License for the file.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val licenseComments: String = "",
+
+    /**
+     * The concluded license for the file as SPDX expression. To represent a not present value [SpdxConstants.NONE] or
+     * [SpdxConstants.NOASSERTION] must be used.
+     */
+    val licenseConcluded: String,
+
+    /**
+     * The license information found in this file. To represent a not present value [SpdxConstants.NONE] or
+     * [SpdxConstants.NOASSERTION] must be used.
+     */
+    val licenseInfoInFiles: List<String>,
+
+    /**
+     * License notices or other such related notices found in the file. This may include copyright statements.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val noticeText: String = ""
+) {
+    enum class Type {
+        /**
+         * The file is associated with a specific application type (MIME type of application e.g. .exe).
+         */
+        APPLICATION,
+
+        /**
+         * Indicates the file is an archive file.
+         */
+        ARCHIVE,
+
+        /**
+         * The file is associated with an audio file (MIME type of audio, e.g. .mp3).
+         */
+        AUDIO,
+
+        /**
+         * Indicates the file is not a text file.
+         */
+        BINARY,
+
+        /**
+         * The file serves as documentation.
+         */
+        DOCUMENTATION,
+
+        /**
+         * The file is associated with an picture image file (MIME type of image, e.g. .jpg, .gif).
+         */
+        IMAGE,
+
+        /**copyrightText
+         * Indicates the file is not a source, archive or binary file.
+         */
+        OTHER,
+
+        /**
+         * Indicates the file is a source code file.
+         */
+        SOURCE,
+
+        /**
+         * The file is an SPDX document.
+         */
+        SPDX,
+
+        /**
+         * The file is a human readable text file (MIME type of text).
+         */
+        TEXT,
+
+        /**
+         * The file is associated with a video file (MIME type of video, e.g. .avi, .mkv, .mp4)
+         */
+        VIDEO;
+    }
+
+    init {
+        require(checksums.filter { it.algorithm == SpdxChecksum.Algorithm.SHA1 }.isNotEmpty()) {
```

Comment:
I guess we could do `checksums.any { it.algorithm == SpdxChecksum.Algorithm.SHA1 }` instead.

Reply:
Done.